docs/design/single-file-unsupported.md

```diff
@@ -0,0 +1,68 @@
+# Single-file unsupported attribute
+
+Starting with .NET Core 3.0 users have the ability to bundle all files used by their applications into a single executable through the publish functionality, `PublishSingleFile`. Although most applications can be agnostic of whether they will be deployed as single-file, there exist some APIs that are known to be incompatible with this deployment mode and thus can cause single-file apps to crash.
+
+While the existing single-file analyzer can help us emit warnings whenever incompatible runtime library APIs are called from within user's code in a project that has been marked with the intent of being published as single-file (through setting `PublishSingleFile` to true in its .csproj), developers can still take a dependency on third-party components implementing functionality that inadvertently make their application no longer compatible with single-file.
+
+## User Experience
+
+For library authors, the new attribute will allow them to easily annotate APIs known to be problematic when used in the context of a single-file application.
+
+```C#
+[SingleFileUnsupported("'Bar' method is not compatible with single-file", "https://help")]
+void Bar()
```

Comment:
It would be better to try to come up with some realistically looking samples - it makes it much easier to read and understand. For example the method could be "GetFileNextToAssembly" or something like that - to make it easy to understand what the function might be about, and also something which is obviously not compatible with single-file at least directly.

Reply:
I've updated the examples given so that it's more clear when would the attribute be used.